src/thor/matrix_action.cc

```diff
@@ -21,43 +21,9 @@ constexpr uint32_t kCostMatrixThreshold = 5;
 namespace valhalla {
 namespace thor {
 
-std::string thor_worker_t::matrix(Api& request) {
-  // time this whole method and save that statistic
-  auto _ = measure_scope_time(request);
-
-  auto& options = *request.mutable_options();
-  adjust_scores(options);
-  auto costing = parse_costing(request);
-
-  // TODO: do this for others as well
-  costmatrix_.set_interrupt(interrupt);
-
-  // lambdas to do the real work
-  auto costmatrix = [&](const bool has_time) {
-    return costmatrix_.SourceToTarget(request, *reader, mode_costing, mode,
-                                      max_matrix_distance.find(costing)->second, has_time,
-                                      options.date_time_type() == Options::invariant,
-                                      options.shape_format());
-  };
-  auto timedistancematrix = [&]() {
-    if (options.shape_format() != no_shape)
-      add_warning(request, 207);
-    return time_distance_matrix_.SourceToTarget(request, *reader, mode_costing, mode,
-                                                max_matrix_distance.find(costing)->second,
-                                                options.matrix_locations(),
-                                                options.date_time_type() == Options::invariant);
-  };
+MatrixAlgorithm* thor_worker_t::get_matrix_algorithm(Api& request, const bool has_time) {
```

Comment:
the diff in here is impossible to read. will show tmrw in code editor what's going on there. the basic gist is:
- add a `get_matrix_algorithm` function to thor_worker_t (like for routing), to get the properties of that class
- remove the lambdas for the matrix algos and decide here which one to use